cosmo_tester/test_suites/test_snapshots/hello_world_snapshot_migration_test.py

```diff
@@ -0,0 +1,326 @@
+            'manager_dir': self._get_manager_dir(version),
+            'activate_path': os.path.join(self._get_activate_path(version)),
+        }
+
+        self._run_script('uninstall_hello_world.sh', arguments)
+
+    def _stop_hello_world(self, version):
+        arguments = {
+            'manager_dir': self._get_manager_dir(version),
+            'activate_path': os.path.join(self._get_activate_path(version)),
+        }
+
+        self._run_script('stop_server.sh', arguments)
+
+    def _assert_server_not_running(self, version):
+        arguments = {
+            'server_url': self._get_server_url_file_path(version)
+        }
+
+        self._run_script('assert_server_not_running.sh', arguments)
+
+    @staticmethod
+    def _get_manager_name(version):
+        return 'manager-{0}'.format(version)
+
+    def _get_manager_dir(self, version):
+        return os.path.join(self.workdir, version.replace('.', ''))
+
+    def _get_venv_dir(self, version):
+        return os.path.join(self._get_manager_dir(version), 'venv')
+
+    def _get_activate_path(self, version):
+        return os.path.join(self._get_venv_dir(version), 'bin', 'activate')
+
+    def _get_snapshots_path(self, version):
+        return os.path.join(self._get_manager_dir(version), 'snapshot')
+
+    def _get_blueprints_dir(self, version):
+        return os.path.join(self._get_manager_dir(version), 'blueprints')
+
+    def _get_server_url_file_path(self, version):
+        return os.path.join(self._get_manager_dir(version), 'webserver_url')
+
+    def _prepare_hello_world_inputs(self):
+        inputs = {
+            'agent_user': self.env.cloudify_agent_user,
+            'image': self.env.ubuntu_trusty_image_name,
+            'flavor': self.env.flavor_name
+        }
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(json.dumps(inputs))
+
+            return f.name
+
+    def _run_script(self, name, arguments):
+        script_path = pkg_resources.resource_filename(
+            cosmo_tester.__name__,
+            'resources/scripts/snapshots_migration_scripts/{0}'.format(
+                name)
+        )
+
+        command = 'bash "{0}"'.format(script_path)
+        rc = subprocess.call(command, shell=True, env=arguments)
+
+        if rc:
+            self.fail('Running script {0} failed (return code = {1})'.format(
+                name, rc))
+
+    def _test_migration(self, source, destination):
+        self._bootstrap_manager(source)
+        self._install_hello_world(source)
+        self._store_server_url(source)
+        self._create_and_download_snapshot(source)
+        self._bootstrap_manager(destination)
+        self._upload_snapshot(source, destination)
+        self._restore_snapshot_and_install_agents(destination)
+        self._assert_server_running(source)
+        self._stop_hello_world(destination)
+        self._uninstall_hello_world(destination)
+        self._assert_server_not_running(source)
```

Comment:
the assertion should come after `stop_hello_world` and before `uninstall_hello_world`